### PR TITLE
fix(deps): update expo monorepo

### PR DESCRIPTION
**Pull request by CraigNewlands**
---------------------

Self Review:

* [ ] Appropriate test coverage
* [ ] Relevant Documentation updated

Smoke Tests:

* [x] Run on ios and update edge
* [x] Run on android and update edge

### CODE REVIEW

File: example/src/useFirmwareUpdate.ts [hunk 1]
```diff
@@ -15,10 +15,20 @@ const useFirmwareUpdate = (
...
+      {
+        estimatedSwapTime: 60,
+        upgradeMode,
+      },
+      (newProgress) => {
+        setProgress(newProgress);
+      },
+      (newState) => {
+        setState(newState);
+      }
```
Comment (ball-hayden) [inline example/src/useFirmwareUpdate.ts:30]: [nitpick]

```suggestion
      setProgress,
      setState,
```

File: react-native-mcu-manager/ios/DeviceUpgrade.swift [hunk 2]
```diff
@@ -171,12 +174,7 @@ extension DeviceUpgrade: FirmwareUpgradeDelegate {
   /// - parameter controller: The controller that may be used to pause,
   ///   resume or cancel the upgrade.
   func upgradeDidStart(controller: FirmwareUpgradeController) {
-    self.manager.updateState(
-      state: [
-        "id": self.id,
-        "state": "STARTED",
-      ]
-    )
+    self.stateHandler(self.id)
```
Comment (ball-hayden) [inline react-native-mcu-manager/ios/DeviceUpgrade.swift:177]:
```suggestion
    self.stateHandler("STARTED")
```

File: react-native-mcu-manager/src/Upgrade.ts [hunk 3]
```diff
@@ -91,16 +97,24 @@ class Upgrade {
...
     bleId: string,
-    updateFileUriString: string,
-    updateOptions: UpgradeOptions
+    updateFileUri: string,
+    updateOptions: UpgradeOptions,
+    private onProgress?: (progress: number) => void,
+    private onStateChange?: (state: string) => void
   ) {
-    this.id = uuidv4() as UpgradeID;
+    this.id = generateUUID() as UpgradeID;
```
Comment (ball-hayden) [inline react-native-mcu-manager/src/Upgrade.ts:105]: Not really any need for this to be in UUID form...

We could switch this out for a "large enough" int?

```suggestion
    this.id = Math.floor(100000000 + Math.random() * 900000000);
```

File: react-native-mcu-manager/src/Upgrade.ts [hunk 4]
```diff
@@ -117,15 +131,11 @@ class Upgrade {
...
-
-        listener(event);
-      }
-    );
+  ): EventSubscription => {
+    return McuManagerEvents.addListener(eventType, (event) => {
+      if (event.id !== this.id) return;
+      listener(event);
+    });
   };
```
Comment (ball-hayden) [inline react-native-mcu-manager/src/Upgrade.ts:139]: With the callback, I don't think this does anything anymore?
Can we drop this?

**Review by ball-hayden — COMMENTED** [review]
We should mark this commit with a `BREAKING CHANGE` footer since we're changing the interface to be callbacks rather than events.

File: react-native-mcu-manager/src/Upgrade.ts (hunk 4 above)
Comment (ball-hayden) [reply react-native-mcu-manager/src/Upgrade.ts:139]: Oh. Sorry. I see now.
Yeah - ignore me, and the comment about the `BREAKING CHANGE` footer.

File: react-native-mcu-manager/src/Upgrade.ts (hunk 3 above)
Comment (CraigNewlands) [reply react-native-mcu-manager/src/Upgrade.ts:105]: Do we still need it to be a string?

File: react-native-mcu-manager/src/Upgrade.ts [hunk 5]
```diff
@@ -91,16 +89,24 @@ class Upgrade {
...
     bleId: string,
-    updateFileUriString: string,
-    updateOptions: UpgradeOptions
+    updateFileUri: string,
+    updateOptions: UpgradeOptions,
+    private onProgress?: (progress: number) => void,
+    private onStateChange?: (state: string) => void
   ) {
-    this.id = uuidv4() as UpgradeID;
+    this.id = String(Math.floor(100000000 + Math.random() * 900000000)) as UpgradeID;
```
Comment (ball-hayden) [inline react-native-mcu-manager/src/Upgrade.ts:97]: Just mind the lint warning here.
Otherwise :+1:

**Review by ball-hayden — APPROVED** [review]

**Pull request merged**